### ISSUE TITLE
Default options in yarnc not always overridden by command line args

### ISSUE THREAD

**Issue by chooban** [yarnpkg/yarn#6594]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
I cannot always override settings that are currently in yarnrc by passing them on the command line.

**If the current behavior is a bug, please provide the steps to reproduce.**
https://github.com/chooban/yarnrc-poc

This repo has a very basic package file and a yarnrc. The rc file defaults version bumps to major and no git tagging. If I run `yarn version` I get a major bump and no tags as expected. If I run `yarn version --minor` I get the same behaviour.

If I set yarnrc to use a default bump of minor and then override on the command line with `yarn version --major` then I get a major bump, as expected

**What is the expected behavior?**
I would expect command line options to always override those in the rc file.

**Please mention your node.js, yarn and operating system version.**
Node: 10.2.1
Yarn: 1.10.1
OS: macOS 10.13.6

**Comment by rally25rs** [comment]
By " The rc file defaults version bumps to major" do you mean your .yarnrc contains something like
```
--version.major
```

If so, then when you do `yarn version --minor` you are basically calling `yarn version --major --minor` (setting one flag does not unset the other)

**Comment by chooban** [comment]
That's correct, the yarnrc in the repo has that set in it. I figured it was something like order of arguments being supplied, but I would expect defaults in the config, and overrides on the command line.

Trying it on the command line, it appears that running `yarn version --major --minor` or `yarn version --minor --major` both result in a major bump, but the situation I described with `--version.minor true` in the rc file, and `--major` on the command line does appear to override the rc and results in a major bump. It's a little inconsistent, but I expect that's down to order of checking which flags are set in the code.

**Comment by paxperscientiam** [comment]
I would like to add a similar observation.

Condition: CLI arguments in `.yarnrc`, execution from project tree, no environment variables
Command: `yarn add package` => PROJECT_ROOT/node_modules/package
Expectation: as expected

Condition: `--modules-folder vendor/node_modules` in `.yarnrc, execution from project tree, no environment variables
Command: `yarn add package` => ~/vendor/node_modules
Expectation: Package installed into PROJECT_ROOT/vendor/node_modules

Condition: CLI arguments in `.yarnrc`, execution from project tree, no environment variables
Command: `yarn add --modules-folder vendor/node_modules package` => PROJECT_ROOT/vendor/node_modules/package
Expectation: as expected


yarn version 1.12.1
node version v10.13.0
